lib/puppet/provider/ec2_vpc_routetable/v2.rb

```diff
@@ -31,18 +31,20 @@ def self.prefetch(resources)
 
   def self.route_to_hash(region, route)
     ec2 = ec2_client(region)
-    gateway = if route.gateway_id == 'local'
-      'local'
-    else
-      begin
-        igw_response = ec2.describe_internet_gateways(internet_gateway_ids: [route.gateway_id])
-        name_from_tag(igw_response.data.internet_gateways.first)
-      rescue Aws::EC2::Errors::InvalidInternetGatewayIDNotFound
+    if route.state == 'active'
```

Comment:
This looks like something we can query for when making the call to `describe_route_tables`, rather than client side.

http://docs.aws.amazon.com/AWSEC2/latest/CommandLineReference/ApiReference-cmd-DescribeRouteTables.html

Any reason that we're not doing that?